examples-gallery/two-pendulum-cart-variable-gallery.py

```diff
@@ -0,0 +1,311 @@
+)
+# bounding h > 0 herlps avoid 'solutions' with h < 0.
+bounds = {F: (-150., 150.), q1: (-5, 5), h: (1.e-5, 1.0)} 
+
+# %%
+# Create an optimization problem and solve it.
+prob = Problem(obj, 
+               obj_grad, 
+               EOM, 
+               state_symbols, 
+               num_nodes, 
+               interval_value,
+               known_parameter_map=par_map,
+               instance_constraints=instance_constraints,
+               time_symbol=t,
+               bounds=bounds)
+
+# Initial guess.
+initial_guess =  np.zeros(prob.num_free)
+initial_guess[-1] = 0.01
+
+# allows to change the number of iterations.
+# standard is 3000
+prob.add_option('max_iter', 6000)
+
+# Find the optimal solution.
+# iterating and using the solution of iteration i
+# as initial guess for iteration i+1 sometimes helps.
+for i in range(2):
+    solution, info = prob.solve(initial_guess)
+    initial_guess = solution
+    
+    print(f'{i + 1}st iteration: value of objective function is {obj(solution):.2f}')
+    prob.plot_objective_value()
+    print('message from optimizer:', info['status_msg'])
+    print('Iterations needed',len(prob.obj_value))
+    print(f"objective value {info['obj_val']:.3e}")
+    print('h value:', solution[-1])
+    print('\n')
+
+# %%
+# Plot the accuracy of the solution, and the state trajectories.
+fehler = info['g']
+anzahl = len(state_symbols)
+fig, ax = plt.subplots(2*anzahl + 1, 1, figsize=(10, 5*(anzahl+1)), sharex=True)
+duration = (num_nodes - 1) * solution[-1]
+times = np.linspace(0.0, duration, num=num_nodes)
+
+for i, j in enumerate(state_symbols):
+    if j in final_state_constraints.keys():
+        farbe1 = 'red'
+    else:
+        farbe1 = 'black'
+
+    if i < len(state_symbols):
+        farbe = 'b'
+    else:
+        farbe = 'r'
+    ax[i].plot(times, fehler[i * num_nodes:(i + 1) * num_nodes], color = farbe, lw=0.5)
+    ax[i].grid()
+    ax[i].set_ylabel(f'${str(j)}$', color=farbe1)
+ax[-1].set_xlabel('time')
+ax[0]. set_title(f'Errors in the generalized coordinates and speeds \n coordinates/speeds in the final_state_constraints are red', color = 'b');
+print(f'error in the final state constraint q2:  {fehler[-5]:.3e}')
+print(f'error in the final state constraint q3:  {fehler[-4]:.3e}')
+print(f'error in the final state constraint u1:  {fehler[-3]:.3e}')
+print(f'error in the final state constraint u2:  {fehler[-2]:.3e}')
+print(f'error in the final state constraint u3:  {fehler[-1]:.3e}')
+
+#fig, ax = plt.subplots(7, 1, figsize=(10, 15), sharex=True)
+#times = np.linspace(0.0, duration, num=num_nodes)
+
+for i, j in enumerate(state_symbols + specified_symbols):
+    k = anzahl + i
+    ax[k].plot(times, solution[i * num_nodes:(i + 1) * num_nodes])
+    ax[k].grid()
+    ax[k].set_ylabel(str(j))
+ax[-1].set_xlabel('time')
+ax[anzahl]. set_title('Generalized coordinates and speeds')
+ax[-1]. set_title('Control force');
```

Comment:
The plots of the constraints and trajectories are built into opty:

```
Problem.plot_trajectories(solution)
Problem.plot_constraints(solution)
```

Best to use those unless you need some particularly special plot.

Reply:
Problem.plot_trajectories(solution)
the above works fine, but
I get this error:
AttributeError: 'Problem' object has no attribute 'plot_constraints'

Reply:
https://opty.readthedocs.io/stable/api.html#opty.direct_collocation.Problem.plot_constraint_violations

Reply:
> https://opty.readthedocs.io/stable/api.html#opty.direct_collocation.Problem.plot_constraint_violations

Now it works!
On my screen the x axis labelling of the second plot, heading nodes, is difficult to read, reast is fine!

Reply:
The formatting isn't perfect for the default plots, but we can improve that in another PR.

Reply:
So, I did not do anything worng. :-)